network/wsNetwork.go

```diff
@@ -1677,6 +1721,23 @@ func (wn *WebsocketNetwork) OnNetworkAdvance() {
 	wn.lastNetworkAdvanceMu.Lock()
 	defer wn.lastNetworkAdvanceMu.Unlock()
 	wn.lastNetworkAdvance = time.Now().UTC()
+	if !wn.relayMessages {
+		// if we're not a relay, and not participating, we don't need txn pool
+		wasParticipating := atomic.LoadUint32(&wn.isParticipating)
+		isParticipating := wn.node.IsParticipating()
```

Comment:
how often is OnNetworkAdvance called? rather than checking so often whether the current round has a live key, would it be more performant (and simpler) to disable TX at connection time, if no part keys are registered at all (ignoring rounds etc) — and then re-enable them if any part keys are registered later (again ignoring rounds)?

Reply:
Didn't make it event driven, but did make HasLiveKeys more efficient. No copy, just runs through the in-memory cache of participation keys (which is probably empty, or probably returns `true` soon when hitting a valid partkey). The cost from here is taking two locks, but probably not holding them very long.